app/src/main/java/com/lightricks/feedexercise/data/FeedRepository.kt

```diff
@@ -1,9 +1,65 @@
 package com.lightricks.feedexercise.data
 
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.database.FeedItemDBEntity
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.TemplatesMetadata
+import io.reactivex.Completable
+import io.reactivex.Observable
+import io.reactivex.schedulers.Schedulers
+
 /**
  * This is our data layer abstraction. Users of this class don't need to know
  * where the data actually comes from (network, database or somewhere else).
  */
 class FeedRepository {
-    //todo: implement
+    private lateinit var db: FeedDatabase
+    private lateinit var feedApiService: FeedApiService
+    private val urlPrefix = "https://assets.swishvideoapp.com/Android/demo/catalog/thumbnails/"
+    fun setFeedApiService(feedApiService: FeedApiService){
+        this.feedApiService=feedApiService
+    }
+
+    fun setDB(db: FeedDatabase) {
+        this.db=db
+    }
+
+    fun refresh(): Completable {
+        return feedApiService.getFeed()
+            .subscribeOn(Schedulers.io())
+            .flatMapCompletable {
+                updateFromTemplatesMetadata(it)
+            }
+    }
+
+    fun fetchData(): Observable<List<FeedItem>>{
+        return db.feedItemDao().getAll().map { it.toFeedItems() }
+    }
+
+    private fun updateWithFeedItemList(list: List<FeedItem>) :Completable {
+        return db.feedItemDao().deleteAll()
+            .andThen(db.feedItemDao()
+                .insertAll(*list.toFeedItemDBEntities().toTypedArray()))
+    }
+
+    private fun updateFromTemplatesMetadata(list: TemplatesMetadata) :Completable{
```

Comment:
Please, fix the formatting.
`:Completable{` -> `: Completable {`
We have some static code analysis running in the repository, which requires code to be well formatted.